File: backend/src/main/java/cs/vsu/raspopov/carkit/dto/detail/DetailDto.java
package cs.vsu.raspopov.carkit.dto.detail;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.math.BigDecimal;
import java.util.List;

@Getter
@NoArgsConstructor
@SuperBuilder
public class DetailDto {

    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    private String producer;
    private String timeToDelivery;
    private String dimension;
    private BigDecimal count;
    private List<DetailResponse> detailResponses;
    private String detailType;
}

File: frontend/src/app/entity/detail/DetailDto.ts
import { Time } from "@angular/common";
import { DetailResponse } from "./DetailResponse";

export interface DetailDto {
  id: number;
  name: string;
  description: string;
  price: number;
  producer: string;
  timeToDelivery: string;
  dimension: string;
  count: number;
  detailType: string;
  detailResponses: DetailResponse[];
}
